src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,10 +1753,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
```

Comment:
I totally agree for regular refresh token cache entry, and that is what we're doing. For family item, the other reason we don't update cache entry for other client id is, refresh token can record more info, i.e mfa.